app/src/main/java/com/jerboa/api/Http.kt

```diff
@@ -316,9 +316,12 @@ interface API {
 }
 
 sealed class ApiState<out T> {
-    class Success<T>(val data: T) : ApiState<T>()
+
+    abstract class Holder<T>(val data: T) : ApiState<T>()
+    class Success<T>(data: T) : Holder<T>(data)
+    class Awaiting<T>(data: T) : Holder<T>(data) // TODO: need better name for this
```

Comment:
This awaiting always seems to be used for fetching more data / Appending. Might as well call it `Appending`.